### PR TITLE
CLOUDP-87128: Add the inventory of the e2e tests in mongocli

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes
I have added [E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/e2e_test_audit/e2e/E2E-TESTS.md) to keep track of how many commands have an e2e test and updated the PR template.

_Jira ticket:_ CLOUDP-87128

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated `e2e-test-audit.csv` (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: e2e-test-audit.csv
Comment (gssbzn) [inline e2e-test-audit.csv:1]: [q] what are you plans on maintaining this and consider that this is a public repo and we welcome external contributors, this format seems hard to maintain specially when adding new commands?

Comment (gssbzn) [inline e2e-test-audit.csv:1]: also, a file like this should probably be in the `e2e` folder

File: e2e-test-audit.csv
Comment (gssbzn) [inline e2e-test-audit.csv:292]: similar to above, are contributors expected to calculate this when updating this file?

**Review by gssbzn — COMMENTED** [review]
Thanks for adding this, the idea seems promising but in its current format it looks to be adding extra complexity for both maintainers and external contributors and I worry specially about the later, I'd like to look into ways to either automate this file generation or have a simpler format that is easy to update for anyone opening a PR

File: e2e-test-audit.csv
Comment (andreaangiolillo) [reply e2e-test-audit.csv:1]: Given that this file is a `.csv`, it can be opened with any editor and easily updated

File: e2e-test-audit.csv
Comment (andreaangiolillo) [reply e2e-test-audit.csv:292]: Ideally yes, but they probably won’t

File: e2e-test-audit.csv
Comment (colm-quinn) [reply e2e-test-audit.csv:1]: I think the point here is not that it's easy to update, but that it's hard to know what update is needed. Think even if you look at the format in 6 months, will you find it easy to remember what each field does?

If we were using a file input like the above, a son would be more powerful as it can be more expressive and easier to validate.

Comment (gssbzn) [reply e2e-test-audit.csv:1]: even as a `.csv` simple text editors (VS code/Atom/Sublime) may show this in a not tabular (excel) way. I would say [markdown table](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown) could be a better presentation as you could preview the file in your text editor and not need something like excel to make sure you didn't break it

this also goes with my other comment on percentages, I doubt people will take care to update them, so should we keep them? 

I'd like to keep this requirement for an e2e inventory as simple and welcoming for someone not familiar with the codebase as possible so if we need to make changes to how we structure e2e tests to make this inventory even easier I'm ok with that

File: e2e/E2E-TESTS.md
Comment (andreaangiolillo) [inline e2e/E2E-TESTS.md:5]: @gssbzn @colm-quinn Could you tell me what you think about this format? I would like to gather some feedback before adding all the commands

Comment (andreaangiolillo) [reply e2e/E2E-TESTS.md:5]: you can have a look at the file here https://github.com/mongodb/mongocli/blob/e2e_test_audit/e2e/E2E-TESTS.md

Comment (colm-quinn) [reply e2e/E2E-TESTS.md:5]: I like the inventory and markdown table. I'd probably summarise the Y/N steps into a single line so it reads more easily. Something like:

```
- E2E Atlas: 
    -  Possible values: ['Y'|'N']
    - Indicates if an e2e test for the command is present ('Y') or not present ('N') for Atlas
...
```

File: .github/pull_request_template.md
Comment (colm-quinn) [inline .github/pull_request_template.md:36]: [np] It's not a csv anymore, as you're using markdown table.

**Review by gssbzn — APPROVED** [review]
Love this, I'd wait for any comment from @colm-quinn as someone less familiar with the code I think his input could be great here

File: .github/pull_request_template.md
Comment (gssbzn) [reply .github/pull_request_template.md:36]: I would also link to the
```suggestion
- [ ] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
```

Comment (colm-quinn) [reply .github/pull_request_template.md:36]: One last comment I have on the checklist is there is nothing to force the user to update the table. I see we have a code health check. In a follow-up check-in, I think it would be good to add a check that the table has been updated if a new test has been added, otherwise I can see it falling out of sync very easily.

**Comment by colm-quinn** [comment]
> Love this, I'd wait for any comment from @colm-quinn as someone less familiar with the code I think his input could be great here

Just waiting on the final updates and will give this a review - all looking good so far!

File: .github/pull_request_template.md
Comment (andreaangiolillo) [reply .github/pull_request_template.md]: yes, you're right. What I would like to do in a follow-up PR is to automatically update this file. If I cannot achieve it I will add a check.

**Review by gssbzn — APPROVED** [review]
LGTM same condition on letting @colm-quinn have final say

File: e2e/E2E-TESTS.md
Comment (colm-quinn) [inline e2e/E2E-TESTS.md:7]: 💯

**Review by colm-quinn — APPROVED** [review]
LGTM, thanks for those changes, looks great!

File: .github/pull_request_template.md
Comment (colm-quinn) [reply .github/pull_request_template.md]: sgtm 👍

**Pull request merged**
